homeassistant/components/broadlink/remote.py

```diff
@@ -434,3 +438,42 @@ async def _async_learn_rf_command(self, command):
             self.hass.components.persistent_notification.async_dismiss(
                 notification_id="learn_command"
             )
+
+    async def async_delete_command(self, **kwargs):
+        """Delete a list of commands from a remote."""
+        kwargs = SERVICE_DELETE_SCHEMA(kwargs)
+        commands = kwargs[ATTR_COMMAND]
+        device = kwargs[ATTR_DEVICE]
+
+        if not self._state:
+            _LOGGER.warning(
+                "remote.delete_command canceled: %s entity is turned off",
+                self.entity_id,
+            )
+            return
+
+        try:
+            codes = self._codes[device]
+        except KeyError:
+            _LOGGER.error("Failed to delete command. Device not found: %s", device)
+            return
+
+        should_store = False
+
+        for command in commands:
+            try:
+                del codes[command]
+            except KeyError:
+                _LOGGER.error(
+                    "Failed to delete command. Command not found: %s", command
+                )
+            else:
+                should_store = True
+
+        if not codes:
+            del self._codes[device]
+            if self._flags.pop(device, None) is not None:
+                await self._flag_storage.async_save(self._flags)
```

Comment:
Since we are deleting and we use `async_delay_save`?

Reply:
I think so, I'm gonna do some local tests.